.github/workflows/main-pr.yaml

```diff
@@ -129,7 +129,60 @@ jobs:
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go
+      - name: Lint the go files
+        run: |
+          go vet -v ./...
+          go mod tidy
+          git diff --exit-code
       - name: Run unit tests
         run: |
           export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
           make test
+
+  integration-tests:
```

Comment:
The default runner vm does not have sufficient resources to run our integration tests. We are going to need to see about using self-hosted runners.

Reply:
So it appears, although I was hoping 7GB would have been enough.